src/Controller/SantaController.php

```diff
@@ -119,6 +122,55 @@ public function run(MessageDispatcher $messageDispatcher, Rudolph $rudolph, Requ
         return new Response($content);
     }
 
+    public function sendSampleMessage(Request $request, string $application): Response
+    {
+        if (!$request->isMethod('POST') || !$request->isXmlHttpRequest()) {
+            throw $this->createNotFoundException();
+        }
+
+        $application = $this->getApplication($application);
+        // An admin is required to use the sample feature
+        if (!$application->isAuthenticated() || !$application->getAdmin()) {
+            throw $this->createAccessDeniedException();
+        }
+
+        $message = $request->request->get('message');
+
+        $errors = $this->validate([], $message, true);
+
+        if (\count($errors) < 1) {
+            $secretSanta = new SecretSanta(
+                $application->getCode(),
+                $application->getOrganization(),
+                'sample',
+                [],
+                [],
+                $application->getAdmin(),
+                str_replace('```', '', $message)
+            );
+
+            try {
+                $application->sendSecretMessage($secretSanta, $application->getAdmin()->getIdentifier(), $application->getAdmin()->getIdentifier(), true);
+            } catch (MessageSendFailedException $e) {
+                return new JsonResponse([
+                    'success' => false,
+                    'errors' => [
+                        'send' => 'Error when sending the sample message',
+                    ],
+                ]);
+            }
+
+            return new JsonResponse([
+                'success' => true,
+            ]);
+        }
+
+        return new JsonResponse([
+            'success' => false,
+            'errors' => $errors,
+        ]);
```

Comment:
Proper status code is better

Reply:
Js part don't care about the error code, so not sure it's worth it